src/java.base/windows/classes/sun/nio/fs/WindowsPath.java

```diff
@@ -279,6 +289,16 @@ private static boolean isSameDrive(String root1, String root2) {
                Character.toUpperCase(root2.charAt(0));
     }
 
+    // Add long path prefix to path
+    static String addPrefix(String path) {
+        if (path.startsWith("\\\\")) {
+            path = "\\\\?\\UNC" + path.substring(1, path.length());
+        } else {
+            path = "\\\\?\\" + path;
+        }
+        return path;
+    }
+
     // Add long path prefix to path if required
     static String addPrefixIfNeeded(String path) {
         if (path.length() > MAX_PATH) {
```

Comment:
We can change addPrefixIfNeeded to call addPrefix for long path. That would avoid duplicate the code to add the prefix.